pip/compat/__init__.py

```diff
@@ -25,7 +24,11 @@
 ]
 
 
-uses_pycache = hasattr(imp, 'cache_from_source')
+if sys.version_info < (3, 4):
+    import imp
+    uses_pycache = hasattr(imp, 'cache_from_source')
+else:
+    uses_pycache = True
```

Comment:
`imp.cache_from_source` is used in `pip.req.req_uninstall.py`

I think it should be something like this

```
if sys.version_info >= (3, 0):
    uses_pycache = True
    if sys.version_info >= (3, 4):
        from importlib.util import cache_from_source
    else:
        from imp import cache_from_source
else:
     # some distributions of pypy2 used pycache
     # test for this
     import imp
     uses_pycache = hasattr(imp, 'cache_from_source')
     if uses_pycache:
         cache_from_source = imp.cache_from_sources 
```